version: '3.3'

services:


  user:
    build:
      context: .
      dockerfile: ./user_service//Dockerfile
    #networks:
      #- my_network
    depends_on:
      db:
        condition: service_healthy
    ports:
      - "8000:8000"
    environment:
      - DATABASE_URL=postgresql://stud:stud@db/archdb


  server:
    build:
      context: .
      dockerfile: ./conference_service/Dockerfile
    #networks:
      #- my_network
    depends_on:
      - mongo
    ports:
      - "8001:8000"
    environment:
      - MONGO_URL=mongodb://mongo:27017


  db:
    image: postgres:14
    container_name: db
    # networks:
    #   - my_network
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_USER=stud
      - POSTGRES_PASSWORD=stud
      - POSTGRES_DB=archdb
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U stud -d archdb"]
      interval: 5s
      timeout: 5s
      retries: 10
    volumes:
      - postgres-data:/var/lib/postgresql/data
      - ./init-db.sql:/docker-entrypoint-initdb.d/init-db.sql

  mongo:
    image: mongo:5.0
    container_name: mongo
    restart: on-failure
    ports:
      - "27017:27017"
    volumes:
      - mongo-data:/data/db
      - ./mongo-init.js:/docker-entrypoint-initdb.d/mongo-init.js:ro
    healthcheck:
      test: echo 'db.runCommand("ping").ok' | mongosh localhost:27017/test --quiet


  structurizr_my:
    image: structurizr/lite
    container_name: strl1
    ports:
      - 8080:8080
    volumes:
        - ./:/usr/local/structurizr

volumes:
  postgres-data:
  mongo-data:
